vscode-ext/src/extension.ts

```diff
@@ -159,12 +163,23 @@ class ConfigurationProvider implements vscode.DebugConfigurationProvider {
 						// eslint-disable-next-line @typescript-eslint/naming-convention
 						'MIRRORD_AGENT_IMPERSONATED_POD_NAMESPACE': podNamespace,
 						// eslint-disable-next-line @typescript-eslint/naming-convention
+						'MIRRORD_IMPERSONATED_CONTAINER_NAME': containerName,
+						// eslint-disable-next-line @typescript-eslint/naming-convention
 						'MIRRORD_AGENT_NAMESPACE': globalContext.workspaceState.get('agentNamespace', 'default'),
 						// eslint-disable-next-line @typescript-eslint/naming-convention
 						'MIRRORD_FILE_OPS': globalContext.workspaceState.get('fileOps', 'false')
 					}
 				};
 				config.env[environmentVariableName] = path.join(libraryPath, libraryName);
+
+				// let user select container name if there are multiple containers in the pod
+				const pod = pods.body.items.find(p => p.metadata!.name === podName!);
```

Comment:
Working on this part still. I think the bug is I shouldn't be nesting these quickselect windows. There's an issue with the program not starting debug session after selecting the container name. I'll keep working on this but sending draft PR in case anyone notices an obvious bug in this part.